scio-core/src/main/scala/com/spotify/scio/coders/Coder.scala

```diff
@@ -262,26 +262,27 @@ private[scio] case class WrappedBCoder[T](u: BCoder[T]) extends BCoder[T] {
         // prior to scio 0.8, a wrapped exception was thrown. It is no longer the case, as some
         // backends (e.g. Flink) use exceptions as a way to signal from the Coder to the layers
         // above here; we therefore must alter the type of exceptions passing through this block.
-        throw CoderStackTrace.append(ex, None, materializationStackTrace)
+        throw CoderStackTrace.append(ex, materializationStackTrace)
     }
 
   override def encode(value: T, os: OutputStream): Unit =
     catching(u.encode(value, os))
 
+  override def encode(value: T, os: OutputStream, context: BCoder.Context): Unit =
+    catching(u.encode(value, os, context))
+
   override def decode(is: InputStream): T =
     catching(u.decode(is))
 
+  override def decode(is: InputStream, context: BCoder.Context): T =
+    catching(u.decode(is, context))
+
```

Comment:
`WrappedBCoder` was not delegating the deprecated `encode` and `decode` methods to the underlying coder.
This can cause issue, eg with the `ProtoCoder`: context is important for the coder, and if not provided, `Context.NESTED` is used. However, when parsing a top level element in a DoFn, `Context.OUTER` must be used instead.

Since `ProtoCoder` is a `org.apache.beam` coder, it wasn't wrapped before. So the fix to reduce size graph has a side effect to hid the `WrappedBCoder` bug